client/src/components/Contents/contentItem.js

```diff
@@ -0,0 +1,30 @@
+import Component from "../../core/component.js";
+import Item from "./item.js";
+
+export default class ContentItem extends Component{
+    template(){
+        return `
+            <li class="contentItem">
+                <h2 class="categoryName">${this.$props.key}</h2>
+                <div class="category">
+                    ${this.$state && Object.keys(this.$state).map(()=>`<Item data-component="Item"></Item>`).join('')}
+                </div>
+            </li>
+        `
+    }
+    mount(){
+        if(!this.$state){
+            const newState=[];
+            for(let i=0;i<4;i++){
+                newState.push(this.$props.value[i]);
+            }
+            this.setState(newState);
+        }
+        else{
+            const $Item=this.$target.querySelectorAll('[data-component="Item"]');
+            for(let i=0;i<$Item.length;i++){
+                new Item($Item[i],{items:this.$state,item:this.$state[i],api:this.$props.api});
+            }
+        }
```

Comment:
```suggestion
        const { $state, $props } = this;
        if(!.$state){
            const newState = Array(4).fill(0).map((v, k) => $props.value[k]);
            this.setState(newState);
            return;
        }
        this.$target.querySelectorAll('[data-component="Item"]').forEach(v => {
            new Item(v, { items: this.$state, item: v, api: this.$props.api });
        });
```
이런식으로 배열 관련 메소드를 적극적으로 사용해보면 어떨까요?
props 에 this.$state를 전부 넘기는건 비효율적인 것 같습니다.
api 부분도 마찬가지입니다.